content/livesync/models.textile

```diff
@@ -0,0 +1,356 @@
+
+h2(#create-or-retrieve). Create or retrieve a model
+
+Instantiate a model using a unique channel name using the @models.get()@ method on the models client. A unique channel name identifies the model on the client and corresponds to the "channel":/channels name used to subscribe to state updates from the backend. If the channel name already exists, then the existing channel will be returned:
+
+```[javascript]
+const model = modelsClient.models.get({
+  channelName: 'posts:123',
+  sync,
+  merge,
+});
+```
+
+TODO: Diagram of how the model components work together
+
+h2(#create-sync). Create a Sync function
+
+The sync function is used by the Models SDK to fetch the latest data from your backend, the SDK will automatically call this function when it is initialized, and when the SDK detects that the latest data is not longer available on the Ably Channel.
+
+The function can contain any custom functionality within it, but needs to return a promise with two objects within the response. The first object is a field called a "@sequenceId@":#sequence-id, which is used to identify the point in the stream of change events that corresponds to the current version of the database's state. The second object is the data relevant to your expected data model.
+
+The following is an example sync function that fetches the latest model state from your database by calling the @/api/posts@ endpoint in your backend:
+
+```[javascript]
+async function sync() {
+  // A call to your backend webserver to retrieve the current state of a post with the ID 123 within your database.
+  const { sequenceId, data } = await fetch('/api/posts/123')
+
+  return { sequenceId, data }
+}
+```
+
+The following is an example output from the sync function i.e. a direct result returned from your endpoint:
+
+```[json]
+{
+  // The latest sequenceId corresponding to the current version of the database's state.
+  "sequenceId": '1',
+  // The contents of data relevant to your expected model. In this instance it is a model of a post with the ID 123.
+  'data':
+    {
+      'id': 123,
+      'text': 'Hello World',
+      'comments': []
+    }
+}
+```
+
+The SDK automatically attempts to re-execute the sync function in the case of an error, as specified by the @syncOptions.retryStrategy@ defined in your "@ClientOptions@":#client-options. To configure a @retryStrategy@ set an integer value of milliseconds for the time between retry attempts. Set this value to @-1@ to stop automatic retries.
+
+h3(#sequence-id). Sequence ID
+
+The @sequenceId@ is returned by the Database Connector along with the data that has been updated. This lets the Models SDK to identify the point in the stream of change events that corresponds to the current version of the database's state.
```

Comment:
I think this wording is wrong. The `sequenceId `is returned by the users sync function, not the Database Connector. The sync function will likely be calling some sort of API in which the sequenceId comes from the users backend.

It is returned along with the latest state at the point in time the sequenceId references.

The database connector doesn't expose any apis and is never interacted with by the models SDK or clients app.

Reply:
Correct - and the sequence ID is obtained from the user's backend sync endpoint by querying the outbox table